tiatoolbox/models/architecture/sccnn.py

```diff
@@ -0,0 +1,372 @@
+                out_width (int):
+                    Output Width
+
+            Returns:
+                tuple of :class:`torch.Tensor`:
+                    Row, Column and height estimates used for
+                    spatially_constrained_layer2 mapping.
+
+            """
+            sigmoid = layer["conv1"](in_tensor)
+            sigmoid0 = sigmoid[:, 0:1, :, :] * (out_height - 1)
+            sigmoid1 = sigmoid[:, 1:2, :, :] * (out_width - 1)
+            sigmoid2 = sigmoid[:, 2:3, :, :]
+            return sigmoid0, sigmoid1, sigmoid2
+
+        def spatially_constrained_layer2(network, sc1_0, sc1_1, sc1_2) -> torch.Tensor:
+            """Spatially constrained layer 2.
+
+            Estimates row, column and height for sc2 layer mapping.
+
+            Args:
+                network (:class:`.SCCNN`):
+                    An initiated SCCNN class.
+                sc1_0 (torch.Tensor):
+                    Output of spatially_constrained_layer1 estimating
+                    the x position of the nucleus.
+                sc1_1 (int):
+                    Output of spatially_constrained_layer1 estimating
+                    the y position of the nucleus.
+                sc1_2 (int):
+                    Output of spatially_constrained_layer1 estimating
+                    the confidence in nucleus detection.
+
+            Returns:
+                :class:`torch.Tensor`:
+                    Probability map using the estimates from
+                    spatially_constrained_layer1.
+
+            """
+            x = torch.tile(
+                network.x, dims=[sc1_0.size(0), 1, 1, 1]
+            )  # Tile for batch size
+            y = torch.tile(network.y, dims=[sc1_0.size(0), 1, 1, 1])
+            xvr = (x - sc1_0) ** 2
+            yvc = (y - sc1_1) ** 2
+            out_map = xvr + yvc
+            out_map_threshold = torch.lt(out_map, network.radius).type(torch.float32)
+            denominator = 1 + (out_map / 2)
+            sc2 = sc1_2 / denominator
+            return sc2 * out_map_threshold
+
+        l1 = conv_act_branch(self.layer["l1"], in_tensor=input_tensor)
+        p1 = self.layer["pool1"](l1)
+        l2 = conv_act_branch(self.layer["l2"], in_tensor=p1)
+        p2 = self.layer["pool1"](l2)
+        l3 = conv_act_branch(self.layer["l3"], in_tensor=p2)
+        l4 = conv_act_branch(self.layer["l4"], in_tensor=l3)
+        drop1 = self.layer["dropout1"](l4)
+        l5 = conv_act_branch(self.layer["l5"], in_tensor=drop1)
+        drop2 = self.layer["dropout2"](l5)
+        s1_sigmoid0, s1_sigmoid1, s1_sigmoid2 = spatially_constrained_layer1(
+            self.layer["sc"], drop2
+        )
+        return spatially_constrained_layer2(self, s1_sigmoid0, s1_sigmoid1, s1_sigmoid2)
+
+    @staticmethod
+    def postproc(
+        prediction_map: np.ndarray, min_distance: int = 6, threshold_abs: float = 0.10
+    ) -> np.ndarray:
+        """Post-processing script for MicroNet.
```

Comment:
Please add a little bit of explanation of what is happening in the function, if possible.

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/434/commits/994dff485177c239529a7a7636cb257adbf50390